examples/src/bin/simpleserver.rs

```diff
@@ -9,27 +9,30 @@
 //! Note that `unwrap()` is used to deal with networking errors; this is not something
 //! that is sensible outside of example code.
 
+use std::env;
 use std::error::Error as StdError;
 use std::fs::File;
 use std::io::{BufReader, Read, Write};
 use std::net::TcpListener;
 use std::sync::Arc;
 
-const CERTFILE: &str = match option_env!("CERTFILE") {
-    Some(certfile) => certfile,
-    None => "localhost.pem",
-};
-
-const PRIV_KEY_FILE: &str = match option_env!("PRIV_KEY_FILE") {
-    Some(priv_key_file) => priv_key_file,
-    None => "localhost-key.pem",
-};
-
 fn main() -> Result<(), Box<dyn StdError>> {
-    let certs = rustls_pemfile::certs(&mut BufReader::new(&mut File::open(CERTFILE)?))
+    let mut args = env::args();
+    let cert_file = match args.next() {
+        Some(certfile) => certfile,
+        None => "localhost.pem".into(),
+    };
+
+    let private_key_file = match args.next() {
```

Comment:
Not sure it really makes sense to allow one argument (which will, from my reading, be the certificate file name with the private key taken from the "localhost-key.pem" default). I'd be in favour of dropping the defaults altogether, and just requiring two arguments.

Reply:
Good point. I've simplified.